konnectivity-client/pkg/client/client.go

```diff
@@ -95,7 +95,31 @@ func (t *grpcTunnel) serve(c clientConn) {
 	defer c.Close()
 
 	for {
-		pkt, err := t.stream.Recv()
+		// It is possible for stream.Recv() to block indefinitely, especially when the stream is full.
+		// Workaround is to run stream.Recv() in a separate goroutine with a generous timeout to prevent
+		// goroutine leaks. See https://github.com/grpc/grpc-go/issues/1229 for more details.
+		done := make(chan struct{})
+		pktChan := make(chan *client.Packet, 1)
+		errChan := make(chan error, 1)
+		go func() {
+			pkt, err := t.stream.Recv()
+			pktChan <- pkt
+			errChan <- err
+			close(done)
```

Comment:
This should probably be a defer close(done) at the beginning of the anonymous function.